checks/shell_download_validate.go

```diff
@@ -316,6 +323,111 @@ func isExecuteFiles(node syntax.Node, cmd, pathfn string, files map[string]bool,
 	return ok
 }
 
+func isGoUnpinnedDownload(cmd []string) bool {
+	if len(cmd) == 0 {
+		return false
+	}
+
+	if !isBinaryName("go", cmd[0]) {
+		return false
+	}
+
+	// `Go install` will automatically look up the
+	// go.mod and go.sum, so we don't flag it.
+	l := 2
```

Comment:
why do we need this var? I see this is used both here and on line 358 in an overloaded way.

in general for constants, better to do:

```go
const descriptiveName = 2
```

in this case though, I think we don't need this and can just explain this with a comment

```go
if len(cmd) < 2 {
  // either `go install` (which checks go.sum) or an unknown command.
  return false
}
```

Reply:
I added this variable because the linter complains that's `2` is an arbitrary number. I'll add a `//nolint` instead.